src/extensions/web/Microsoft.DotNet.UpgradeAssistant.Steps.Razor/RazorSourceUpdater.cs

```diff
@@ -207,6 +214,8 @@ private async Task<IEnumerable<MappedTextReplacement>> GetReplacements(Project o
 
         private static async Task FixUpProjectFileAsync(IUpgradeContext context, CancellationToken token)
         {
+            // Reload the workspace in case code fixes modified the project file
+            await context.ReloadWorkspaceAsync(token).ConfigureAwait(false);
```

Comment:
The plenty of `ReloadWorkspaceAsync` we have around the project is affecting perf of the tool especially where bigger solutions are involved. We need to look at ways for an alternate here.

Reply:
Agreed. I think that's out-of-scope for this change, but we should review them at some point.